lib/prometheus/middleware/exporter.rb

```diff
@@ -0,0 +1,93 @@
+# encoding: UTF-8
+
+require 'prometheus/client'
+require 'prometheus/client/formats/text'
+
+module Prometheus
+  module Middleware
+    # Exporter is a Rack middleware that provides a sample implementation of a
+    # Prometheus HTTP exposition endpoint.
+    #
+    # By default it will export the state of the global registry and expose it
+    # under `/metrics`. Use the `:registry` and `:path` options to change the
+    # defaults.
+    class Exporter
+      attr_reader :app, :registry, :path
+
+      FORMATS  = [Client::Formats::Text].freeze
+      FALLBACK = Client::Formats::Text
+
+      def initialize(app, options = {})
+        @app = app
+        @registry = options[:registry] || Client.registry
+        @path = options[:path] || '/metrics'
+        @acceptable = build_dictionary(FORMATS, FALLBACK)
+      end
+
+      def call(env)
+        if env['PATH_INFO'] == @path
+          format = negotiate(env['HTTP_ACCEPT'], @acceptable)
+          format ? respond_with(format) : not_acceptable(FORMATS)
+        else
+          @app.call(env)
+        end
+      end
+
+      private
+
+      def negotiate(accept, formats)
+        accept = '*/*' if accept.to_s.empty?
+
+        parse(accept).each do |content_type, _|
+          return formats[content_type] if formats.key?(content_type)
+        end
+
+        nil
+      end
+
+      def parse(header)
+        header.to_s.split(/\s*,\s*/).map do |type|
```

Comment:
This method is used only by `negotiate` and we will never pass `header` as non-string value, so we can ommit `to_s` here